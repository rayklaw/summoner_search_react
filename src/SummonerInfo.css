.ProfileIcon {
    position: relative;
}
.ProfileIcon > img{
    width: 120px;
    height: 120px;
    float: left;
    margin-left: 390px;
}

.TierBorder > img {
    position: absolute;
    width: 140px;
    height: 140px;
    float: left;
    top: 190px;
    margin-left: -130px;
}

.SummonerLevel {
    position: absolute;
    margin-top: 110px;
    margin-left: -80px;
    width: 44px;
    height: 24px;
    padding-top: 3px;
    box-sizing: border-box;
    background: url(http://opgg-static.akamaized.net/images/site/summoner/bg-levelbox.png);
    background-size: 100%;
    line-height: 17px;
    font-family: Helvetica,AppleSDGothic,"Apple SD Gothic Neo",AppleGothic,Arial,Tahoma;
    font-size: 14px;
    text-align: center;
    color: #eabd56;
}

.Name {
    position: relative;
    font-weight: bold;
    font-size: 50px;
    margin-left: 50px;
    top: 30px;
}
