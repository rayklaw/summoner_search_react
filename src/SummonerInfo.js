import React from 'react'

import './SummonerInfo.css'

const SummonerInfo = props => {
  const { summoner } = props
  const { rankedStats } = props

  let soloIndex;

  rankedStats.forEach((stat, index) => {
    if (stat.queueType === "RANKED_SOLO_5x5") {
      soloIndex = index;
    }
  })

  return (
    <div className="SummonerInfo">
      <div className="ProfileIcon">
        <img src={`http://ddragon.leagueoflegends.com/cdn/9.22.1/img/profileicon/${summoner.profileIconId}.png`} alt="" />
      </div>
      <div className="TierBorder">
        {soloIndex !== undefined && <img src={`https://opgg-static.akamaized.net/images/borders2/${rankedStats[soloIndex].tier.toLowerCase()}.png`} alt="" />}
      </div>
      <span className="SummonerLevel">{summoner.summonerLevel}</span>
      <span className="Name">{summoner.name}</span>
    </div>
  )
}

export default SummonerInfo