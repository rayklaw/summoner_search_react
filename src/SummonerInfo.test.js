import React from 'react'
import ReactDOM from 'react-dom'
import TestRenderer from 'react-test-renderer'
import ReactTestUtils from 'react-dom/test-utils'

import sinon from 'sinon'

import SummonerInfo from './SummonerInfo'

import * as mock from './mock'

it('summoner info renders', () => {
    let rankedStats = [{
        "leagueId": "d42df200-3018-11e9-9bdb-c81f66cf135e",
        "queueType": "RANKED_SOLO_5x5",
        "tier": "CHALLENGER",
        "rank": "I",
        "summonerId": "NtW1fFWiqG2Nqw5fo1NaUeI4ouvXXcHFT4hz6VsrEy5eSsE",
        "summonerName": "MyAir",
        "leaguePoints": 64,
        "wins": 4,
        "losses": 20,
        "veteran": false,
        "inactive": false,
        "freshBlood": false,
        "hotStreak": false
    }]
    let summoner = {
        "id": "bJ3fnl4axekkZNRW8i0r-DzCOltAQCMilbmhHFPo-HL-dkE",
        "accountId": "_ETokaUkycTk1_xuYLmiMaE8mGl0GQUihhMWSDmW7WNk2g",
        "puuid": "eVRErZLpnKEFBUEVLcMuAh9QmhJ_lWPjIpYjosFjCgu18Ikz_1czfw5QIWadEao2610GsbwkUar8_g",
        name: "MyAir",
        profileIconId: 588,
        revisionDate: 1571275176000,
        summonerLevel: 98
    }
    const component = TestRenderer.create(<SummonerInfo rankedStats={rankedStats} summoner={summoner} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})