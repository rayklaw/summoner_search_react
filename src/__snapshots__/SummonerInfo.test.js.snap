// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summoner info renders 1`] = `
<div
  className="SummonerInfo"
>
  <div
    className="ProfileIcon"
  >
    <img
      alt=""
      src="http://ddragon.leagueoflegends.com/cdn/9.22.1/img/profileicon/588.png"
    />
  </div>
  <div
    className="TierBorder"
  >
    <img
      alt=""
      src="https://opgg-static.akamaized.net/images/borders2/challenger.png"
    />
  </div>
  <span
    className="SummonerLevel"
  >
    98
  </span>
  <span
    className="Name"
  >
    MyAir
  </span>
</div>
`;
